upload/helpers.py

```diff
@@ -806,13 +807,23 @@ def generate_upload_sentry_metrics_tags(
         endpoint=endpoint,
         is_using_shelter="yes" if is_shelter_request else "no",
     )
+
+    repo_visibility = None
     if repository:
-        metrics_tags["repo_visibility"] = (
-            "private" if repository.private is True else "public"
-        )
-    if position:
-        metrics_tags["position"] = position
-    if upload_version:
-        metrics_tags["upload_version"] = upload_version
+        repo_visibility = "private" if repository.private else "public"
+
+    optional_fields = {
+        "repo_visibility": repo_visibility,
+        "position": position,
+        "upload_version": upload_version,
+    }
```

Comment:
In the original code, the `optional_fields` are not created if we don't provide them in the arguments, and python's `prometheus-client` errors if we initialize a metric with N labels, but call them with some empty labels. This change allows it to create these fields that are filled with None such that they are present in the labels dict.

The original code does not create the fields if the corresponding arguments are not provided. However, the prometheus-client in Python raises an error if we initialize a metric with a certain number of labels but then call it with some empty labels. I introduced `optional_fields` dict here to ensure that these fields are always present in the labels dictionary, even if they are filled with None.

i.e. old code, `generate_tags(..., endpoint="some_endpoint")` returns:
```
{
   ...
   endpoint: "some_endpoint",
}
```
in the new code, it will return
{
   ...
   endpoint: "some_endpoint",
   repo_visibility: "None",
   position: "None",
   upload_version: "None",
}

If we still want the first dict to be returned, we can call generate tags with `fill_labels=False`.

Reply:
I checked and that's fine, in prometheus it will interpret it as a string `"None"`.